compute-budget/src/compute_budget_processor.rs

```diff
@@ -110,7 +110,10 @@ pub fn process_compute_budget_instructions<'a>(
                     if updated_loaded_accounts_data_size_limit.is_some() {
                         return Err(duplicate_instruction_error);
                     }
-                    updated_loaded_accounts_data_size_limit = Some(bytes);
+                    updated_loaded_accounts_data_size_limit = Some(
+                        NonZeroU32::new(bytes)
+                            .ok_or(TransactionError::InvalidLoadedAccountsDataSizeLimit)?,
+                    );
```

Comment:
This is where the new error handling has been added. After this change, any call to `process_compute_budget_instructions` in the tx pipeline will return an error if the loaded accounts data size limit is 0

Reply:
Do we need a feature gate for this? TX sets data size limit to 0 would be OK in before-code, but will error now.

Reply:
I'm in the process to refactor `process_compute_budget_instructions()`, break it into two parts: one part that it *only* process cb ixs, read whatever user set in TX, so its result is deterministic per tx. 2nd part to sanitize/validate into ComputeBudgetLimits, which take sworking_bank/feature_set.

Reply:
A feature gate shouldn't be needed for this, if you see something that indicates otherwise, let me know! This PR has the same observable side effect as before.. the transaction just gets dropped. As for your refactor, that sounds like it could be nice! `process_compute_budget_instructions` is quite a beast right now

Reply:
Ah right, this change moves same error `InvalidLoadedAccountsDataSizeLimit` from `load_accounts()` up earlier to `validate_fees()`, in the same thread; it won't impact consensus, no need to feature gate.